### PR TITLE
feat(CTA): add initial vite support with `vue` and `vue-ts`

### PR DESCRIPTION
**Pull request by amrbashir**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [X] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by amrbashir** [comment]
does this PR need a changefile ?

**Comment by amrbashir** [comment]
Any future vite template will only have to include a folder under `templates/vite` with the modified content.
The folder name must be a name that is one of vite's [official templates](https://github.com/vitejs/vite/tree/main/packages/create-app) for ex: `vanilla-ts`, `react` or `lit-element-ts`

**Comment by jbolda** [comment]
> does this PR need a changefile ?

I think it makes sense to have one. It's a new "feature" so might as well highlight it.

File: tooling/create-tauri-app/src/recipes/react.ts [hunk 1]
```diff
@@ -24,12 +24,14 @@ const afterCra = async (cwd: string, appName: string, version: string) => {
...
   descriptiveName: "React.js",
   shortName: "reactjs",
-  configUpdate: ({ cfg }) => ({
+  configUpdate: ({ cfg, packageManager }) => ({
     ...cfg,
     distDir: `../build`,
     devPath: "http://localhost:3000",
-    beforeDevCommand: `npm start`,
-    beforeBuildCommand: `npm build`,
+    beforeDevCommand: `${packageManager === "yarn" ? "yarn" : "npm run"} start`,
```
Comment (jbolda) [inline tooling/create-tauri-app/src/recipes/react.ts:31]: oh good call 🎉

**Comment by jbolda** [comment]
> Any future vite template will only have to include a folder under `templates/vite` with the modified content.
> The folder name must be a name that is one of vite's [official templates](https://github.com/vitejs/vite/tree/main/packages/create-app) for ex: `vanilla-ts`, `react` or `lit-element-ts`

Do you think there is value in having extra template files for every vite template?

**Review by nothingismagick — APPROVED** [review]

**Comment by amrbashir** [comment]
> Do you think there is value in having extra template files for every vite template?

No not really. The template files are modified to just add the tauri logo next to vue .
If you can do without that then there is no need to have these files.

I wanted to add an example showing the usage of some tauri's api, maybe just a simple communication between js and rust but that can quickly be like the `examples` repo

**Comment by jbolda** [comment]
> > Do you think there is value in having extra template files for every vite template?
> 
> No not really. The template files are modified to just add the tauri logo next to vue .
> If you can do without that then there is no need to have these files.
> 
> I wanted to add an example showing the usage of some tauri's api, maybe just a simple communication between js and rust but that can quickly be like the `examples` repo

Well, it is probably worthwhile to get this in and we can adjust later. I wonder if maybe we can do some sort of "common" template files that all of the examples can use (or at least within the same framework / group) since it probably does make sense long term to show the API use in some manner.

**Review by jbolda — APPROVED** [review]

**Pull request merged**